packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -362,12 +417,36 @@ export class Canary extends cdk.Resource {
     };
   }
 
-  private createRunConfig(props: CanaryProps): CfnCanary.RunConfigProperty | undefined {
-    if (!props.environmentVariables) {
+  private createVpcConfig(props: CanaryProps): CfnCanary.VPCConfigProperty | undefined {
+    if (!props.vpc) {
       return undefined;
     }
+
```

Comment:
I think we should fail synth if a user tries to set vpc related options without setting `vpc`. Right now they look to be just ignored in that case, which might be cause for confusion.

Reply:
I think the two places I took for inspiration don't explicitly do this, _but_ I think it's a good idea.